xbmc/interfaces/legacy/Control.cpp

```diff
@@ -1055,6 +1055,16 @@ namespace XBMCAddon
       return msg.GetLabel();
     }
 
+    void ControlEdit::setType(int type, const String& heading)
+    {
+      if (pGUIControl)
+      {
+        LOCKGUI;
+
+        ((CGUIEditControl *) pGUIControl)->SetInputType((CGUIEditControl::INPUT_TYPE)type, CVariant{heading});
```

Comment:
I agree with you on this one.
this would be `static_cast<CGUIEditControl*>(pGUIControl)->` and same for the type